lib/CMakeLists.txt

```diff
@@ -30,7 +30,9 @@ add_subdirectory_ifdef(CONFIG_SUPL_CLIENT_LIB supl)
 add_subdirectory_ifdef(CONFIG_DATE_TIME date_time)
 add_subdirectory_ifdef(CONFIG_EDGE_IMPULSE edge_impulse)
 add_subdirectory_ifdef(CONFIG_WAVE_GEN_LIB wave_gen)
-add_subdirectory_ifdef(CONFIG_HW_UNIQUE_KEY hw_unique_key)
+if (DEFINED CONFIG_HW_UNIQUE_KEY OR DEFINED CONFIG_HW_UNIQUE_KEY_LOAD)
```

Comment:
IMO we should use a separate kconfig symbol for this, which is selected by both `CONFIG_HW_UNIQUE_KEY` and `CONFIG_HW_UNIQUE_KEY_LOAD` so that we only need to perform a single `add_subdirectory_ifdef(CONFIG_HW_UNIQUE_KEY_ANY)`.

Only a nit, so leave this unless you need to do some more pushing to this branch :)